source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -2229,21 +2229,33 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
             inst->removeAndDeallocate();
 
         // Translate types.
-        List<IRHLSLStructuredBufferTypeBase*> instsToProcess;
-        List<IRInst*> textureFootprintTypes;
+        List<IRHLSLStructuredBufferTypeBase*> buffersToLower;
+        List<IRMatrixType*> matricesToLower;
```

Comment:
My preference is that matrix type lowering should be hoisted into its own pass that is called when we are emitting code for spirv/glsl, instead of being a part of the spirv-legalize pass.